pkg/hubble/relay/observer.go

```diff
@@ -88,14 +95,47 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 			}
 		})
 	}
+
+	pq := queue.NewPriorityQueue(qlen)
+	sortedFlows := make(chan *observerpb.GetFlowsResponse, qlen)
+	go func() {
+		defer close(sortedFlows)
+	flowsLoop:
+		for {
+			select {
+			case flow, ok := <-flows:
+				if !ok {
+					break flowsLoop
+				}
+				if pq.Len() == qlen {
+					f := pq.Pop()
+					select {
+					case sortedFlows <- f:
+					case <-ctx.Done():
+						return
+					}
+				}
+				pq.Push(flow)
+			case <-time.After(1 * time.Second): // make sure to drain the queue when no new flow responses are received
```

Comment:
nit: Since we are calling this for each flow and it is presumably creating a channel each time, I wonder if creating a `time.NewTimer` and calling `Reset` at the end of the loop might be slightly more efficient. But I actually don't know, just something that stood out to me.

Reply:
I could add a benchmark in the tests and actually measure this.